ui/src/components/SampleGrid/SampleCircleView.jsx

```diff
@@ -0,0 +1,119 @@
+    );
+    event.stopPropagation();
+  }
+
+  function isPuckSelected(cellID, puckID) {
+    return (
+      Number(filterOptions.cellFilter) === cellID &&
+      Number(filterOptions.puckFilter) === puckID
+    );
+  }
+
+  useEffect(() => {
+    if (filterOptions.cellFilter === '') {
+      dispatch(filterAction({ cellFilter: `${1}`, puckFilter: `${1}` }));
+    }
+  }, [filterOptions.cellFilter, dispatch]);
+
+  function handleDisplayPuckCellContextMenu(e, menuID, cellID, puckID) {
+    e.preventDefault();
+    handleClickOnCellPuck(e, cellID, puckID === null ? '' : puckID);
+    displayPuckCellContextMenu(e, menuID, cellID, puckID);
+    e.stopPropagation();
+  }
+
+  function getSingleCellAsCircle(centerX = 10, centerY = 10) {
+    const pucks = scContent.children;
+    const totalPucks = pucks.length;
+    const cellID = 1;
+
+    // Adjust angle range based on puck count
+    const useFullCircle = totalPucks >= 8;
+    const angleRange = useFullCircle ? 2 * Math.PI : Math.PI * 1.5;
+    const startAngle = useFullCircle ? 0 : (Math.PI - angleRange) / 2;
+
+    // Circle size and spacing
+    const baseRadius = 4;
+    const placementRadius = baseRadius + Math.max(0, (totalPucks - 16) * 0.3);
+    const puckRadius = Math.min(
+      0.7,
+      (2 * Math.PI * placementRadius) / (totalPucks * 2),
+    );
+
+    // Fix angle spacing: avoid skipping puck 1 or overlapping at end
+    const angleDivisor = useFullCircle
+      ? totalPucks
+      : Math.max(1, totalPucks - 1);
+
+    return pucks.map((_, i) => {
+      const angle = startAngle + (angleRange * i) / angleDivisor;
+      const x = centerX + placementRadius * Math.cos(angle);
+      const y = centerY + placementRadius * Math.sin(angle);
+      const puckID = i + 1;
+
+      return (
+        <g
+          key={`single_cell_puck_${puckID}`}
+          onClick={(e) => handleClickOnCellPuck(e, cellID, puckID)}
+          onContextMenu={(e) =>
+            handleDisplayPuckCellContextMenu(e, puckMenuID, cellID, puckID)
+          }
+          fill={isPuckSelected(cellID, puckID) ? '#015f9d' : '#cdced1'}
+          className="puck_cicle"
+        >
+          <circle
+            cx={x}
+            cy={y}
+            r={puckRadius}
+            stroke="#01011ba2"
+            strokeWidth="0.05"
+          />
+          <text
+            x={x}
+            y={y}
+            fontSize="0.3"
+            textAnchor="middle"
+            dominantBaseline="middle"
+            fill={isPuckSelected(cellID, puckID) ? 'white' : 'black'}
+          >
+            Puck {puckID}
+          </text>
+          <title>
+            Cell: {cellID}, Puck: {puckID}
+          </title>
+        </g>
+      );
+    });
+  }
+
+  return (
+    <Col sm={6}>
+      <div className="div-svg-flex">
+        <svg height="100%" width="100%" viewBox="0 1 20 20">
+          <circle fill="#6cb0f5" r="8" cx="10" cy="10" />
+          {getSingleCellAsCircle()}
```

Comment:
Just to reiterate, I would prefer for the two arguments to be passed explicitly here instead of having default values.